packages/astro/components/Image.astro

```diff
@@ -0,0 +1,28 @@
+---
+import { getImage, type LocalImageProps, type RemoteImageProps } from 'astro:assets';
+import { AstroError, AstroErrorData } from '../dist/core/errors/index.js';
+
+// The TypeScript diagnostic for JSX props uses the last member of the union to suggest props, so it would be better for
+// LocalImageProps to be last. Unfortunately, when we do this the error messages that remote images get are complete nonsense
+// Not 100% sure how to fix this, seems to be a TypeScript issue. Unfortunate.
+type Props = LocalImageProps | RemoteImageProps;
```

Comment:
Does `RemoteImageProps['src']` always have a protocol? Is it possible to brand an interface that way so that `http(s)://`-prefixed `src` indicates we should use `RemoteImageProps`? I guess not because we don’t have a way to make `Props` take a type argument?

Reply:
It does for our services (http(s), ftp(s) and data:), but other services might use `src` for any kind of keys, so we can't do that 😔